### PR TITLE
Fixes #106, #115

### LINKED ISSUE #106: Create command to turn paths and intersections into a graph

### PR DESCRIPTION
**Pull request by versae**
Fixes #106, #115
- Adding a command, `graph`, to build a graph from a black and white image grid and a list of regions as returned by the `matches` command
- Each edge keeps information about its entire path and a polyline simplified version as well
- The generated graph can be saved in several formats
- Support for Python 3.4 has been dropped due to the simplification library being only available for Python 3.5 and 3.6

### CODE REVIEW

**Comment by mejackreed** [comment]
The code looks good to me. One thing to note is that our Python deployment only has python 3.4 currently. See: https://github.com/sul-dlss/puppet/search?utf8=%E2%9C%93&q=pip3&type=

I'm not sure if it is difficult or not to upgrade to 3.5 or 3.6

cc @eefahy

File: histonets/api.py [hunk 1]
```diff
@@ -397,3 +401,38 @@ def histogram_palette(histogram, n_colors=8, method='auto', sample_fraction=5,
...
+
+@image_as_array
+def extract_edges(image, regions, simplification_method=None,
+                  simplification_tolerance=1.0):
+    """Build a graph from edges expressed in a binary grid (0's representing
+    holes, 1's paths), and vertices as the center of regions expressed by the
+    the top-left corner and the down-right corner in (x, y) pixels coordinates
+    from the grid."""
+    if simplification_method == 'vw':
+        simplify_func = simplify_coords
```
Comment (csbailey5t) [inline histonets/api.py]: Should these be reversed? It seems like you're checking for vw, if it's set, use the basic simplify_coords. And if it's not, then use simplify_coords_vw.  If the simplification method is vw, should you use the simplify_coords_vw?

Comment (versae) [reply histonets/api.py]: And you're right. Good catch!

**Comment by versae** [comment]
@mejackreed our other option is to use [simplify.py](https://github.com/omarestrella/simplify.py), the Python version of [simplify.js](http://mourner.github.io/simplify-js/) (extracted from Leaflet by Vladimir Agafonkin). It only supports a combination of RDP with Radial Distance, so Visvalingam-Wyatt algorithm would not be available (and it's faster for the worst case). In terms of performance, since `simplification` is a Rust library, is orders of magnitude faster than `simplify.py`, although it might still work for our case.

Let's see what @eefahy thinks and the decide.

File: histonets/utils.py [hunk 2]
```diff
@@ -593,3 +616,136 @@ def sample_histogram(histogram, sample_fraction=0.05):
...
+                if border_horizontal:
+                    point = border_x + border_step, border_y
+                else:
+                    point = border_x, border_y + border_step
+                inner_paths[draw.line(*point[::-1], *center[::-1])] = color
+    return inner_paths.tocsc()
+
+
+def edges_to_graph(edges, fmt=None):
+    """Buid a graph based on a list of edges and serialize it to format. Each
```
Comment (csbailey5t) [inline histonets/utils.py]: Typo in "build"

**Comment by versae** [comment]
After conversations with @eefahy it seemed unlikely to setup support for 3.5 in our VMs. However, @urschrei, the creator of the [`simplification`](https://github.com/urschrei/simplification) library, kindly agreed to re-release his library with support for 3.4 under a `manylinux1` wheel 💃  (see https://github.com/urschrei/simplification/issues/1).

I'll clean up the commit history and force push a new rebase.

File: histonets/utils.py (hunk 2 above)
Comment (versae) [reply histonets/utils.py]: Fixed.

**Review by csbailey5t — APPROVED** [review]

**Pull request merged**